plugins/modules/web_infrastructure/jira.py

```diff
@@ -149,6 +150,32 @@
     default: true
     type: bool
 
+  attachment:
+    type: dict
+    required: false
+    aliases: [file]
+    description:
+      - Information about the attachment being uploaded
+    suboptions:
+      filename:
+        required: true
+        type: str
```

Comment:
Being a file path, I would rather go with 
```suggestion
        type: path
```
instead.